test/integration/conftest.py

```diff
@@ -21,8 +21,14 @@ def tmp_dir_and_file(ee_contents):
 
 
 @pytest.fixture()
-def ee_tag():
-    return str(uuid.uuid4())[:10]
+def ee_tag(request, cli, container_runtime):
+    image_name = 'builder-test-' + str(uuid.uuid4())[:10]
+
+    def delete_image():
+        cli(f'{container_runtime} rmi {image_name}')
+
+    request.addfinalizer(delete_image)
+    return image_name
```

Comment:
ooo looks good to me

Reply:
actually thinking about, this may act weird because `container_runtime` is parametrized....

Reply:
I really don't think this will cause any problems. To test this, let me show you:

```diff
diff --git a/test/integration/conftest.py b/test/integration/conftest.py
index f54ac52..b35a42d 100644
--- a/test/integration/conftest.py
+++ b/test/integration/conftest.py
@@ -22,6 +22,7 @@ def build_dir_and_ee_yml():
 
 @pytest.fixture()
 def ee_tag(request, cli, container_runtime):
+    raise Exception(f'runtime: {container_runtime}')
     image_name = 'builder-test-' + str(uuid.uuid4())[:10]
 
     def delete_image():
@@ -71,4 +72,5 @@ def container_runtime(request, cli):
     if cli(f'{request.param} --version', check=False).returncode == 0:
         return request.param
     else:
+        return request.param
         pytest.skip(f'{request.param} runtime not available')
```

that gives:

```
=================================================================================== ERRORS ===================================================================================
____________________________________________________________ ERROR at setup of test_build_streams_output[docker] _____________________________________________________________

request = <SubRequest 'ee_tag' for <Function test_build_streams_output[docker]>>, cli = <function cli.<locals>.run at 0x10e668950>, container_runtime = 'docker'

    @pytest.fixture()
    def ee_tag(request, cli, container_runtime):
>       raise Exception(f'runtime: {container_runtime}')
E       Exception: runtime: docker

test/integration/conftest.py:25: Exception
____________________________________________________________ ERROR at setup of test_build_streams_output[podman] _____________________________________________________________

request = <SubRequest 'ee_tag' for <Function test_build_streams_output[podman]>>, cli = <function cli.<locals>.run at 0x10e943a70>, container_runtime = 'podman'

    @pytest.fixture()
    def ee_tag(request, cli, container_runtime):
>       raise Exception(f'runtime: {container_runtime}')
E       Exception: runtime: podman

test/integration/conftest.py:25: Exception
___________________________________________________________ ERROR at setup of test_user_system_requirement[docker] ___________________________________________________________

request = <SubRequest 'ee_tag' for <Function test_user_system_requirement[docker]>>, cli = <function cli.<locals>.run at 0x10e943cb0>, container_runtime = 'docker'

    @pytest.fixture()
    def ee_tag(request, cli, container_runtime):
>       raise Exception(f'runtime: {container_runtime}')
E       Exception: runtime: docker

test/integration/conftest.py:25: Exception
___________________________________________________________ ERROR at setup of test_user_system_requirement[podman] ___________________________________________________________

request = <SubRequest 'ee_tag' for <Function test_user_system_requirement[podman]>>, cli = <function cli.<locals>.run at 0x10e9439e0>, container_runtime = 'podman'

    @pytest.fixture()
    def ee_tag(request, cli, container_runtime):
>       raise Exception(f'runtime: {container_runtime}')
E       Exception: runtime: podman

test/integration/conftest.py:25: Exception
```

So I think it's calling the fixture with all the right stuff in all the right places.

The only concern that I personally had is that we may be deleting an image which was never really created, due to an error, or falling into some case where it never builds anything. Let me demo this out:

```
$ docker rmi foobar
Error: No such image: foobar
$ echo $?
1
$ docker rmi -f foobar
Error: No such image: foobar
$ echo $?
0
```

So yes, I think this command _should_ use the `-f` option. I will push a commit that does that. Otherwise, I don't think there's any holes in this, and I really advocate merging this.